xarray/core/computation.py

```diff
@@ -1005,10 +1007,12 @@ def earth_mover_distance(first_samples,
         func = functools.partial(func, **kwargs)
 
     if vectorize:
-        if meta is None:
-            # set meta=np.ndarray by default for numpy vectorized functions
-            # work around dask bug computing meta with vectorized functions: GH5642
-            meta = np.ndarray
+        if dask == "parallelized" and meta is None:
+            # only basic checks: _apply_blockwise will raise the appropriate errors
+            if output_dtypes is not None and isinstance(output_dtypes, list):
+                # set meta=np.ndarray by default for numpy vectorized functions
+                # work around dask bug computing meta with vectorized functions: GH5642
+                meta = np.ndarray((0, 0), dtype=output_dtypes[0])
```

Comment:
What should meta look like if there are multiple outputs with different dtypes?

Reply:
Currently multiple outputs are not supported with `dask="parallelized"` - this is checked in `_apply_blockwise`. I'm nor super happy with the way this is handled in my PR, but didn't want to copy the error check here, either.

@dcherian: should `meta` be created in `_apply_blockwise` instead (even if `vectorize=False`)?

Reply:
I thought about this but decided against it. We only want to automatically specify `meta` when `vectorize=True` because `np.vectorize` will return a numpy array and that too only to work around that dask bug.
I think we should leave the automatic `meta` inference to dask.